docusaurus/docs/engineering/knowledge-base/development-lifecycle.md

```diff
@@ -4,154 +4,239 @@ sidebar_position: 7
 
 # Development Lifecycle
 
-To look at/download BobaBoard's code, check out our guide to [Setting Up your Coding Environment](/docs/engineering/start-developing/before-you-start).
-You might also be interested in [Git Notes](/docs/engineering/legacy-pages/git-notes)
+> //TODO: Consider making this doc a folder with multiple sections and moving to the Volunteering section
 
-:::tip This seems scarier than it is!
-If you're new to coding and github, the following process might seem scary and overwhelming. Don't worry: it will become second nature very quickly! If you want someone to hold your hand throughout the process, contact Ms. Boba.
-:::
+If you're new to collaborating on a code project, you may be wondering what that process looks like. This page will:
+
+- give an overview of the development lifecycle
+- explain key terminology
+- guide you through the step-by-step process of contributing to BobaBoard's codebase.
+
+This document also includes relevant information about git and GitHub to slowly get you familiar with more technical vocabulary you might run into. For more information on these subjects you can look at our [Git Notes](/docs/engineering/legacy-pages/git-notes.md) page.
 
 :::note
-TODO[Ms.Boba]: Explore whether the new github command line interface can make this process less cumbersome.
+We will use GitHub as our example platform since BobaBoard's code is hosted there. However, there are alternative platforms that serve similar purposes.
 :::
 
-## Development lifecycle overview
+## Overview
 
-At a high level, the git development lifecycle follows these steps:
+So, you've decided to contribute to a project. What happens next?
 
-0. **The developer "forks" the code.** This creates a separate copy of the codebase owned by the developer itself. No matter what changes the developer makes, the original codebase (called `upstream`) will remain unchanged.
-1. **The developer creates a "feature branch" on their "fork".** When the developer first forks the code, they are placed on the official "current version" of the codebase, called `master` (or `main`). Rather than modify `master` directly, the developer then creates a "feature branch" to make their edits on.
+1. **Add a copy of the project's code to your own GitHub account.**
 
-   For example, if the developer is working on adding a "read more" feature, they might create a branch called `read-more`. Creating branches allows a developer to work on many independent features at the same time.
+   > This first step involves getting your own version of the project's code to edit to your heart's content. In technical terms, this means creating a 'fork' of the project you want to contribute to.
```

Comment:
What do you think of:
> The first step is to get your own version of the project's code to edit to your heart's content. In technical terms, you'll create a "fork" the project you want to contribute to and add it to your GitHub account.

Reply:
"and add it to your GH account" seems wrong to me but I dunno if my wording is better

Reply:
Maybe: 
`> The first step is to get a copy of the project's code into your own GitHub account by "forking" it. This will allow you to make as many changes to the code as you desire without impacting the original project.`